wd_helper.au3

```diff
@@ -1000,34 +1003,44 @@ EndFunc   ;==>_WD_ElementOptionSelect
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand)
+Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand, $vParameter = Null)
 	Local Const $sFuncName = "_WD_ElementSelectAction"
-	Local $sNodeName, $vResult
-
+	Local $sNodeName, $vResult, $sScript
 	$sNodeName = _WD_ElementAction($sSession, $sSelectElement, 'property', 'nodeName')
 	Local $iErr = @error
 
 	If $iErr = $_WD_ERROR_Success Then
 		If $sNodeName = 'select' Then
 			Switch $sCommand
-				Case 'value'
-					; Retrieve current value of designated Select element
-					$vResult = _WD_ExecuteScript($sSession, "return arguments[0].value", __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
-					$iErr = @error
-
 				Case 'options'
-					; Retrieve array containing value / label attributes from the Select element's options
-					Local $sScript = "var result =''; var options = arguments[0].options; for (let i = 0; i < options.length; i++) {result += options[i].value + '|' + options[i].label + '\n'} return result;"
+					$sScript = "var result =''; var options = arguments[0].options; for (let i = 0; i < options.length; i++) {result += options[i].value + '|' + options[i].label + '\n'} return result;"
 					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
 					$iErr = @error
+
 					If $iErr = $_WD_ERROR_Success Then
 						Local $sText = StringStripWS($vResult, $STR_STRIPTRAILING)
 						Local $aOut[0][2]
 						_ArrayAdd($aOut, $sText, 0, Default, @LF, 1)
 						$vResult = $aOut
 					EndIf
+
+				Case 'selectedIndex'
+					$sScript = "return arguments[0].selectedIndex" & ((IsInt($vParameter)) ? (" = " & $vParameter) : (''))
+					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
+					$iErr = @error
+
+				Case 'selectedText'
+					$sScript = "if (arguments[0].selectedIndex > -1) {return arguments[0].options[arguments[0].selectedIndex].text;} else {return '';}"
```

Comment:
I think that you should consider using `arguments[0].selectedOptions` instead of `arguments[0].options`.

- You could drop the check for `(arguments[0].selectedIndex > -1) ` and instead check `(arguments[0].selectedOptions.length)` 
- Better multiselect support

Reply:
Working on.

Reply:
Please check: [Function Header and SELECTEDTEXT refactored](https://github.com/Danp2/WebDriver/pull/175/commits/e2346981080552cc6d2f1919e2ba6541f2c358b3)

Reply:
Can we treat this part as "Resolved conversation" ?